src/haddock/libs/libplots.py

```diff
@@ -633,22 +636,115 @@ def find_best_struct(ss_file, number_of_struct=10):
-        # name = Path(row).name
-        # return f'<a href="../{row}" target="_blank" rel="noreferer">{name}</a>' # noqa: E501
-        return Path(row).name
 
-    table_df = best_struct_df.copy()
+def _ngl_viewer():
+    # http://nglviewer.org/ngl/gallery/
+    ngl_script = ("""
+        <script src="https://cdn.rawgit.com/arose/ngl/v2.1.0/dist/ngl.js">
+        </script>
+        <script>
+            var dialog;
+            var stage;
+
+            document.addEventListener("DOMContentLoaded", function () {
+                // Setup to load data from rawgit
+                NGL.DatasourceRegistry.add(
+                    "data",
+                    new NGL.StaticDatasource(
+                        "//cdn.rawgit.com/arose/ngl/v2.1.0/data/"
+                        )
+                );
+
+                // Create NGL Stage object
+                stage = new NGL.Stage("viewport");
+
+                // Handle window resizing
+                window.addEventListener("resize", function(event){
+                    stage.handleResize();
+                }, false);
+
+            });
+
+            function showStructure(file_name) {
+                dialog = document.getElementById("structureViewerDialog");
+                dialog.showModal();
+                stage.loadFile(file_name).then(function (o) {
+                    o.addRepresentation("cartoon");
+                    o.autoView();
+                    stage.setParameters({backgroundColor: "white"});
+                    stage.handleResize();
+                });
+            }
+        </script>
+    """)
+    ngl_dialog = ("""
+        <body>
+            <dialog id="structureViewerDialog">
+                <div id="viewport" style="width:800px; height:600px;"></div>
+                <form>
+                    <button value="cancel" formmethod="dialog">X</button>
+                </form>
+            </dialog>
+        </body>
+    """)
+    return ngl_script + ngl_dialog
+
+
+def _add_viewers(df):
+    def _generate_download_link(file_name, dl_name):
+        html_code = "&#8595;"  # add icon
+        html_code += "&nbsp;"  # add space
+        html_code += f'<a href="{file_name}" download="{dl_name}">Download</a>'
+        return html_code
+
+    def _generate_view_link(file_name):
+        html_code = "&#x1F441;"  # add icon
+        html_code += "&nbsp;"  # add space
+        event = f'''onClick="showStructure(\'{file_name}\')"'''  # create event
+        html_code += f'<a {event} style="cursor:pointer;">View</a>'  # add link
+        return html_code
+
+    def _format_cell(row):
+        file_name, dl_name = row.split(",")
+
+        # Correct path because after running analyse
+        # files are moved to analysis folder
+        correct_path = f"../{file_name}"
+
+        # Add suffix to download name
+        suffix = Path(file_name).suffix
+        dl_name = dl_name + suffix
+
+        # Generate html code
+        html_code = "<p>"
```

Comment:
The p-tag adds margin above and below making it look different then the other cells. 

Can you use `span` tag instead so each cell has same height?

Reply:
fixed!